make/proto.mk

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/make -f
+
+###############################################################################
+###                                Protobuf                                 ###
+###############################################################################
+
+containerProtoVer=v0.7
+containerProtoImage=tendermintdev/sdk-proto-gen:$(containerProtoVer)
+containerProtoGen=regen-ledger-proto-gen-$(containerProtoVer)
+containerProtoFmt=regen-ledger-proto-fmt-$(containerProtoVer)
+containerProtoGenSwagger=regen-ledger-proto-gen-swagger-$(containerProtoVer)
+
+proto-all: proto-lint proto-format proto-gen proto-check-breaking
+
+proto-gen:
+	@echo "Generating Protobuf files"
+	@if docker ps -a --format '{{.Names}}' | grep -Eq "^${containerProtoGen}$$"; then docker start -a $(containerProtoGen); else docker run --name $(containerProtoGen) -v $(CURDIR):/workspace --workdir /workspace $(containerProtoImage) \
+		sh ./scripts/protocgen.sh; fi
+
+proto-format:
+	@echo "Formatting Protobuf files"
+	@if docker ps -a --format '{{.Names}}' | grep -Eq "^${containerProtoFmt}$$"; then docker start -a $(containerProtoFmt); else docker run --name $(containerProtoFmt) -v $(CURDIR):/workspace --workdir /workspace tendermintdev/docker-build-proto \
+		find .  -name "*.proto" -not -path "./third_party/*" -exec clang-format -i {} \; ; fi
+
+DOCKER_BUF := docker run -v $(shell pwd):/workspace --workdir /workspace bufbuild/buf:1.7.0
+
+proto-lint:
+	@$(DOCKER_BUF) lint --error-format=json
+
+proto-check-breaking:
+	@$(DOCKER_BUF) breaking --against https://github.com/cheqd/cheqd-node.git#branch=main
+
+GOGO_PROTO_URL           = https://raw.githubusercontent.com/regen-network/protobuf/cosmos
+GOOGLE_PROTO_URL         = https://raw.githubusercontent.com/googleapis/googleapis/master
+COSMOS_PROTO_URL         = https://raw.githubusercontent.com/cosmos/cosmos-sdk/v0.45.4/proto/cosmos
+COSMOS_ORM_PROTO_URL     = https://raw.githubusercontent.com/cosmos/cosmos-sdk/orm/v1.0.0-alpha.10/proto/cosmos
```

Comment:
Pinned at older versions and would need to be bumped and/or replaced with the correct ones.